Display custom Apps' logos

lkostrowski opened the pull request:
Displays logo from Manifest (available from 3.14) in apps list, app installation and app details

- [x] Fix paddings / sizes
- [x] QA scenario
- [x] First merge new schema in https://github.com/saleor/saleor-dashboard/pull/3751

To be tested with App with logo:
https://saleor-app-products-feed-git-app-logo-saleorcommerce.vercel.app/api/manifest

or any preview URLs from this PR https://github.com/saleor/apps/pull/552#issuecomment-1587529079
(copy preview URL + add `/api/manifest`

QA:
1. Using app that does not contain `brand.logo.default` field in manifest will not display custom logo - will fallback to generic icon 
2. Using app with `brand.logo.default` will use this logo in following places: app installation page, app list - during installation, app list - after installation, app details

Other changes
1. App installation page icons are changed: new Saleor logo + rectangle icons instead circles
2. App icon size in app details page is bigger

### Screenshots



| | |
|--------|--------|
|<img width="1171" alt="Zrzut ekranu 2023-06-13 o 09 58 16" src="https://github.com/saleor/saleor-dashboard/assets/9268745/192df2ee-f340-420e-95db-479855399de6"> | <img width="1173" alt="Zrzut ekranu 2023-06-13 o 09 58 10" src="https://github.com/saleor/saleor-dashboard/assets/9268745/c8639ef5-700f-4640-8a1b-f05c0500b207"> |
| <img width="750" alt="Zrzut ekranu 2023-06-13 o 09 57 57" src="https://github.com/saleor/saleor-dashboard/assets/9268745/8a2ef2f0-c47d-4cc6-bd78-dcb330742f1f"> | <img width="754" alt="Zrzut ekranu 2023-06-13 o 09 57 52" src="https://github.com/saleor/saleor-dashboard/assets/9268745/625bb632-8fcf-453d-b6a8-7c761ce5c311"> |
| <img width="1151" alt="Zrzut ekranu 2023-06-13 o 09 57 36" src="https://github.com/saleor/saleor-dashboard/assets/9268745/c3ad6156-3082-45a9-bb86-66b5d55363f0"> | <img width="1152" alt="Zrzut ekranu 2023-06-13 o 09 57 29" src="https://github.com/saleor/saleor-dashboard/assets/9268745/a0fcb1d5-065b-479c-9437-94437a8c810f"> |













### Pull Request Checklist

<!-- Please keep this section. It will make maintainer's life easier. -->

1. [x] This code contains UI changes
4. [x] All visible strings are translated with proper context including data-formatting
5. [ ] Attributes `data-test-id` are added for new elements
6. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
7. [x] Your code works with the latest stable version of the core


### Test environment config

<!-- Do not remove this section. It is required to properly setup test deployment instance.
Modify API_URI if you want test instance to use custom backend. CYPRESS_API_URI is optional, use when necessary. -->

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
3. [ ] giftCard
4. [ ] category
5. [ ] collection
6. [ ] attribute
8. [ ] productType
9. [ ] shipping
10. [ ] customer
11. [ ] permissions
12. [ ] menuNavigation
13. [ ] pages
14. [ ] sales
15. [ ] vouchers
16. [ ] homePage
17. [ ] login
18. [ ] orders
19. [ ] products
20. [x] app
21. [ ] plugins
22. [ ] translations
23. [ ] navigation
24. [ ] variants
25. [ ] payments

CONTAINERS=1

krzysztofwolski commented on src/apps/components/AppInstallPage/AppInstallPage.tsx:
```diff
@@ -39,6 +41,18 @@ export const AppInstallPage: React.FC<AppInstallPageProps> = ({
   const classes = useStyles();
+  const { theme } = useTheme();
+
+  const getSaleorLogoUrl = () => {
```
We already have a themeable Saleor logo in the side navigation. Is the code duplicated, or was there another reason?

lkostrowski replied:
Previously it was using a fixed image. In this diff I replaced it to one that is used in the navigation. Its not themable - these are 2 different PNGs

wojteknowacki commented:
App img during the installation process is not visible on the app list view from the start - default img is visible and then a new one is placed in its place or sometimes new img is not replacing the default one at all

https://github.com/saleor/saleor-dashboard/assets/124166231/4783bb37-e496-48f9-a880-ea2ae96eea1f

wojteknowacki commented:
Not all apps placeholders images are visible on the list view after installation - for example slack 
![Zrzut ekranu 2023-06-13 o 16 49 31](https://github.com/saleor/saleor-dashboard/assets/124166231/c71926f6-e1ab-42fe-a7d5-d39efc87b05a)
Not sure is it related to this PR changes or not

lkostrowski commented:
> There is only one view that does not have an app img - Manage App - is it on purpose? ![Zrzut ekranu 2023-06-13 o 16 28 45](https://user-images.githubusercontent.com/124166231/245509640-1b8709ce-f817-4d67-b429-1f24fd82824c.png)

It was never there, but its a good time add it 👍

lkostrowski commented:
@wojteknowacki build fill fail - I have to fix tests, but:

1. I used the sam navbar in Manage App page now
2. I fixed not loading Slack and other "official apps" icons. They will fallback to generic defaults. Its on purpose, because once apps are released, source o images will be changed. In this PR it will be regression but on the release it will be invisible for the user
3. Delayed loading of images is known issue. I added separate ticket for it https://github.com/saleor/apps/issues/562. I think its minor. When used refresh pages/navigate back, logos will be loaded. Its caused by Saleor generating logos async

lkostrowski commented on src/apps/components/AppDetailsPage/Header.test.tsx:
Removed test. It tests nothing and is very flaky

lkostrowski commented on src/fragments/apps.ts:
```diff
@@ -17,6 +17,11 @@ export const appManifestFragment = gql`
       code
       name
     }
+    brand {
+      logo {
+        default
+      }
+    }
```
TIL :) I will update it

lkostrowski replied:
fixed
